router/sources/router.move

```diff
@@ -363,6 +365,14 @@ module router::router {
             );
             let token_id = aptos_names::token_helper::latest_token_id(&tokendata_id);
 
+            // Domain must migrate before subdomain, throw error if this is a subdomain but domain has not been migrated
+            if (option::is_some(&subdomain_name)) {
+                assert!(
+                    exists_in_v2(domain_name, option::none()) && is_name_owner(user_addr, domain_name, option::none()),
```

Comment:
I think this error would apply even if the user is not owner of the domain. Do we need the `is_name_owner` check?

Reply:
i think user must be the domain owner to be able to migrate the subdomain, @angieyth @BriungRi i think so, right?

Reply:
I  think Renee means we should throw the error when the domain is not v2 (whether the user is the owner or not)

Reply:
but also I think you don't have to be the domain owner to migrate the subdomain(?) you need to be the subdomain owner to do so.

Reply:
+1 to @angieyth, for example I created a domain and subdomain, then transferred the subdomain to someone else. They should have the power to migrate the subdomain.

Reply:
make sense, updated